spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -333,6 +352,14 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 					error = ((ErrorMessage) reply).getPayload();
 				}
 			}
+			if (reply == null && this.errorOnTimeout) {
```

Comment:
I don't really understand what you mean; if the service activator invokes a method that returns `void` or a method that returns `null`, the flow ends at that point and the header enricher will never be invoked. This is not a change in the current version - it has always worked that way.

This change simply returns a 5xx error if no reply is received within the timeout - it doesn't change the behavior of the downstream flow.

If you want to add a header when using an inbound channel adapter, an alternative would be to add a `<filter/>` to the `WEB-XML`.